### PR TITLE
Template generation

### PR DESCRIPTION
**Pull request by iocanel**
Resolves: #12

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Generate Backstage Catalog Information as part of the Quarkus build or the Quark
 
 - Generate the catalog-info.yaml for the Quarkus application
 - Command Line interface to install / uninstall and Component to Backstage
+- Generate Backstage Template from an existing Quarkus application
```
Comment (cmoulliard) [inline README.md:11]: Generate a Backstage Template for ....
Remark: You should here explain a bit what the template will do

**Review by cmoulliard — CHANGES_REQUESTED** [review]
Build of the PR fails => to be fixed please 

```
[INFO] Quarkus Backstage - Deployment ..................... FAILURE [  0.051 s]
[INFO] Quarkus Backstage - CLI ............................ SKIPPED
[INFO] Quarkus Backstage - Documentation .................. SKIPPED
[INFO] Quarkus Backstage - Integration Tests .............. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.879 s
[INFO] Finished at: 2024-10-10T09:28:36+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project quarkus-backstage-deployment: Could not resolve dependencies for project io.quarkiverse.backstage:quarkus-backstage-deployment:jar:999-SNAPSHOT: The following artifacts could not be resolved: io.quarkiverse.backstage:quarkus-backstage-rest-client:jar:999-SNAPSHOT (absent): Could not find artifact io.quarkiverse.backstage:quarkus-backstage-rest-client:jar:999-SNAPSHOT -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :quarkus-backstage-deployment
```

File: README.md (hunk 1 above)
Comment (iocanel) [reply README.md:11]: Doc updated with more details about the generated template. However, I respectfully disagree on the `s/from/for` thing. We are not generating a template `for` an application as we do for example with the `catalog-info.yaml`. We are using the application itself as input and thus `from` is more appropriate in this case.

File: README.md [hunk 2]
```diff
@@ -107,3 +110,64 @@ To list all entitties installed
 ```shell
 quarkus backstage entities list
 ```
+
+### Templates
+
+#### Generating a Backstage Template
+
+The backstage extension is able to generate a backstage template from an existing Quarkus application.
+The generated template will include a parmeterized version of the project and a template definition.
```
Comment (cmoulliard) [inline README.md:119]: s/parmeterized/parameterized

File: README.md [hunk 3]
```diff
@@ -107,3 +110,64 @@ To list all entitties installed
...
+quarkus backstage template list
+```
+
+The command will generate a template under the `.backstage/templates` directory.
+The template can then be manually imported to backstage.
+
+
+#### Installing a Backstage Template
+
+The generated template can be installed to backstage using the following command:
```
Comment (cmoulliard) [inline README.md:161]: Has the user using the Quarkus CLI or extension the same RBAC as users accessing the backstage UI ?

Comment (iocanel) [reply README.md:161]: Yes

**Pull request merged**